### PR TITLE
Add Compose interview

### PR DESCRIPTION
**Pull request by brian-brazil**

### CODE REVIEW

File: content/blog/2016-09-21-interview-with-compose.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+their own metrics on each host, but by combining it with Telegraf, we can
+arrange to have each host export metrics for all its containers (as well as its
+overall resource metrics) via a single Prometheus scrape target.
+
+
+## How did you transition?
+
+We are a Chef shop so we spun up a largish instance with a big EBS volume and
+then reached right for a [community chef
+cookbooks](https://github.com/rayrod2030/chef-prometheus) for Prometheus.
```
Comment (RichiH) [inline content/blog/2016-09-21-interview-with-compose.md]: "a cookbooks"?

File: content/blog/2016-09-21-interview-with-compose.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+We have a lot of containers on each host so we were expecting to have to start
+to shard Prometheus once we added all memory usage metrics from those too, but
+Prometheus just kept on going without any drama and still without getting too
+close to saturating its resources. We currently monitor over 400,000 distinct
+metrics every 15 seconds for around 40,000 containers on 450 hosts with a
+single m4.xlarge prometheus instance wtih 1TB of storage. You can see our host
+dashboard for this host below. Disk IO on the 1TB gp2 SSD EBS volume will
+probably be the limiting factor eventually. Our initial guess is well
+over-provisioned for now but we are growing fast both metrics gathered and
```
Comment (RichiH) [inline content/blog/2016-09-21-interview-with-compose.md]: That sentence seems broken.

File: content/blog/2016-09-21-interview-with-compose.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+We also have several other systems that we want to get into Prometheus to get
+better visibility. We run our apps on Mesos and have integrated basic Docker
+container metrics already, which is better than previously, but we also want to
+have more of the infrastructure components in the Mesos cluster recording to
+central Prometheus so we can have centralised dashboards showing all elements
+of supporting system health from load balancers right down to app metrics.
+
+Eventually we will need to shard Prometheus. We already split customer
+deployments among many smaller clusters for a variety of reasons so the one
+logical option would be to move to a smaller Prometheus server (or pair for
```
Comment (RichiH) [inline content/blog/2016-09-21-interview-with-compose.md]: a pair

File: content/blog/2016-09-21-interview-with-compose.md (hunk 1 above)
Comment (brian-brazil) [inline content/blog/2016-09-21-interview-with-compose.md]: Done

File: content/blog/2016-09-21-interview-with-compose.md (hunk 2 above)
Comment (brian-brazil) [inline content/blog/2016-09-21-interview-with-compose.md]: Done

File: content/blog/2016-09-21-interview-with-compose.md (hunk 3 above)
Comment (brian-brazil) [inline content/blog/2016-09-21-interview-with-compose.md]: Done

**Comment by RichiH** [comment]
Looking good to me.

Richard

Sent by mobile; excuse my brevity.

**Comment by RichiH** [comment]
PS: It might be a good idea to pace those.

File: content/blog/2016-09-21-interview-with-compose.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+fall over. It didn’t! In fact it handled the load of host-level metrics scraped
+every 15 seconds for around 450 hosts across our newer infrastructure with very
+little resource usage.
+
+We have a lot of containers on each host so we were expecting to have to start
+to shard Prometheus once we added all memory usage metrics from those too, but
+Prometheus just kept on going without any drama and still without getting too
+close to saturating its resources. We currently monitor over 400,000 distinct
+metrics every 15 seconds for around 40,000 containers on 450 hosts with a
+single m4.xlarge prometheus instance wtih 1TB of storage. You can see our host
```
Comment (juliusv) [inline content/blog/2016-09-21-interview-with-compose.md]: wtih -> with

File: content/blog/2016-09-21-interview-with-compose.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+resource usage for our own operations. The next logical step is enabling the
+database teams to push metrics to the local Telegraf instance from inside the
+DB containers so we can record database-level stats too without increasing
+number of targets to scrape.
+
+We also have several other systems that we want to get into Prometheus to get
+better visibility. We run our apps on Mesos and have integrated basic Docker
+container metrics already, which is better than previously, but we also want to
+have more of the infrastructure components in the Mesos cluster recording to
+central Prometheus so we can have centralised dashboards showing all elements
```
Comment (juliusv) [inline content/blog/2016-09-21-interview-with-compose.md]: "to central" -> "to the central"

**Comment by juliusv** [comment]
:+1: Great interview!

The only thing making me slightly uneasy is that it could inspire others to follow the practice of using uber-exporters, and the reasoning for having them (like having fewer scrape targets - shouldn't be a problem, but a pro?). But not sure if we can remark anything to that effect.

**Comment by fabxc** [comment]
You can add a comment to the blog post once it's up :)
Ultimately, we cannot prevent people from doing what they want to do. We
should focus on providing guidance and features (i.e. great integrations
with K8S SD for example) that makes people not feel like they need/want it.

File: content/blog/2016-09-21-interview-with-compose.md (hunk 5 above)
Comment (brian-brazil) [inline content/blog/2016-09-21-interview-with-compose.md]: Done

File: content/blog/2016-09-21-interview-with-compose.md (hunk 4 above)
Comment (brian-brazil) [inline content/blog/2016-09-21-interview-with-compose.md]: Done

**Pull request merged**